### ISSUE TITLE
Argument order for locks.acquire

### ISSUE THREAD

**Issue by pwnall** [inexorabletash/web-locks#19]
(while it may be worse for WebIDL) I think it'd be better if the options bag preceded the callback. JS code is easier to read when functions take at most one callback, which is the last argument.

The example below is lifted from the explainer. Baseline:
```javascript
  await locks.acquire('resource', async lock => {
    // Use |lock| here.
  }, {signal: controller.signal});
```

This proposal:
```javascript
  await locks.acquire('resource', {signal: controller.signal}, async lock => {
    // Use |lock| here.
  });
```

So, please reconsider the argument order.

**Comment by inexorabletash** [comment]
@domenic - what do you think?

We've got two conflicting precedents here - putting callbacks last, and putting options last.

**Comment by domenic** [comment]
Yeah, I noticed this before. I guess I do lean toward callback-last. Preferably with an overload so you can omit the options, I guess.

**Comment by inexorabletash** [comment]
Are dictionaries and callbacks distinguishable in WebIDL parlance? [overloading](https://heycam.github.io/webidl/#idl-overloading) still hurts my brain...

i.e. is this valid WebIDL ?

```webidl
interface LockManager {
  Promise<any> acquire((DOMString or sequence<DOMString>) scope,
                       LockRequestCallback callback);                       
  Promise<any> acquire((DOMString or sequence<DOMString>) scope,
                       optional LockOptions options,
                       LockRequestCallback callback);
};
```

**Comment by domenic** [comment]
Well, optional arguments before the end is not valid Web IDL, so that's a minor bug...

They are not distinguishable, but overloads allow you to make decisions based on the number of arguments passed, instead of on distinguishability. This is usually bad practice, but it seems like the best option here...

**Comment by inexorabletash** [comment]
Whoops, copy/pasta.  Removing `optional`

```webidl
interface LockManager {
  Promise<any> acquire((DOMString or sequence<DOMString>) scope,
                       LockRequestCallback callback);                       
  Promise<any> acquire((DOMString or sequence<DOMString>) scope,
                       LockOptions options,
                       LockRequestCallback callback);
};
```

And... I guess argument count trumps distinguishability? I'll give this a whirl in Blink at least.

**Comment by inexorabletash** [comment]
Thanks - yes, code is much more readable this way. Updated the readme, proto spec, idl, and added an FAQ item contrasting the two versions.

**Comment by pwnall** [comment]
🎉 Thank you very much!
